docs/spec/relayer-v2/Chain.tla

```diff
@@ -0,0 +1,146 @@
+---------------------------- MODULE Chain ----------------------------
+
+EXTENDS Naturals, FiniteSets, RelayerDefinitions, 
+        ClientHandlers, ConnectionHandlers, ChannelHandlers
+        
+CONSTANTS ChainID, \* chain identifier
+          Heights \* set of possible heights of the chains in the system 
+
+VARIABLES chainStore,
+          incomingDatagrams
+
+vars == <<chainStore, incomingDatagrams>>          
+
+(***************************************************************************
+ Client update operators
+ ***************************************************************************)
+\* Update the clients on chain with chainID, 
+\* using the client datagrams generated by the relayer      
+\* (Handler operators defined in ClientHandlers.tla)
+LightClientUpdate(chainID, chain, datagrams) == 
+    \* create clients
+    LET clientCreatedChain == HandleCreateClient(chainID, chain, datagrams) IN
+    \* update clients
+    LET clientUpdatedChain == HandleUpdateClient(chainID, clientCreatedChain, datagrams) IN
+
+    clientUpdatedChain
+    
+(***************************************************************************
+ Connection update operators
+ ***************************************************************************)
+\* Update the connections on chain with chainID, 
+\* using the connection datagrams generated by the relayer
+\* (Handler operators defined in ConnectionHandlers.tla)
+ConnectionUpdate(chainID, chain, datagrams) ==
+    \* update chain with "ConnOpenInit" datagrams
+    LET connOpenInitChain == HandleConnOpenInit(chainID, chain, datagrams) IN
```

Comment:
Can order in which we apply those operators matter?I guess that in theory it shouldn't but maybe it could be relevant for some implementations.

Reply:
The output chain of every operator is given as input to the next operator. I think it makes a difference in case there are faulty relayers. If there are `ConnOpenInit` (sent by correct relayer) and `ConnOpenAck` (sent by faulty relayer) in incoming datagrams for `chainA`, and `chainA` is in state `UNINIT`, in this order, after applying `ConnectionUpdate`, the connection end at `chainA` will be in state `OPEN`. If the order of the operators is swapped, then the chain first processes `ConnOpenAck` (discards it), and then processes `ConnOpenInit`, which results in a connection end in state `INIT` at `chainA`.